estate_account/models/estate_property.py

```diff
@@ -0,0 +1,38 @@
+from datetime import date
+from odoo import Command, models
+from odoo.exceptions import UserError
+
+
+class EstateProperty(models.Model):
+    _inherit = "estate.property"
+
+    def action_set_sold(self):
+        result = super().action_set_sold()
+        self.create_invoice()
+        return result
+
+    def create_invoice(self):
+        try:
+            self.check_access_rights('write')
```

Comment:
```suggestion
            self.check_access('write')
```
the function you have used has been deprecated since 18.0